### PR TITLE
Added unit tests for gocql/public/client.go

### PR DESCRIPTION
**Pull request by timl3136**
**What changed?**
Added unit tests for public client and also renamed testBase.go file to testdata.go

**Why?**
Improve unit test coverage and stability

**How did you test it?**
Unit test

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/persistence/nosql/nosqlplugin/cassandra/gocql/public/testdata.go
Comment (taylanisikdemir) [inline common/persistence/nosql/nosqlplugin/cassandra/gocql/public/testdata.go:1]: testdata.go is still in scope of coverage. move it under testdata folder. 
https://github.com/uber/cadence/blob/master/codecov.yml#L43

File: common/persistence/nosql/nosqlplugin/cassandra/gocql/public/client_test.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+// SOFTWARE.
+
+package public
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	gogocql "github.com/gocql/gocql"
```
Comment (taylanisikdemir) [inline common/persistence/nosql/nosqlplugin/cassandra/gocql/public/client_test.go:30]: why rename the import? there's no other `gocql`  here

**Review by taylanisikdemir — APPROVED** [review]
LGTM with comments

File: common/persistence/nosql/nosqlplugin/cassandra/gocql/public/client_test.go (hunk 1 above)
Comment (timl3136) [reply common/persistence/nosql/nosqlplugin/cassandra/gocql/public/client_test.go]: Thanks

**Pull request merged**